File: 119.pascals-triangle-ii.kt
// @leet start
class Solution {
    fun getRow(rowIndex: Int): List<Int> {
      var row = mutableListOf<Int>()
      for (i in 0..rowIndex) {
        var next = mutableListOf<Int>()
        for (j in 0..i) {
          if (j == 0 || j == i) {
            next.add(1)
          } else {
            next.add(row[j] + row[j - 1])
          }
        }
        row = next
      }
      return row
    }
}
// @leet end

File: 119.pascals-triangle-ii.cs
// @leet start
public class Solution {
    public IList<int> GetRow(int rowIndex) {
      var res = new List<int>();
      for (var i = 0; i <= rowIndex; i++) {
        var next = new List<int>();
        for (var j = 0; j <= i; j++) {
          if (j == 0 || j == i) {
            next.Add(1);
          } else {
            next.Add(res[j - 1] + res[j]);
          }
        }
        res = next;
      }
      return res;
    }
}
// @leet end
